packages/run-protocol/src/vpool-xyk-amm/swap.js

```diff
@@ -3,14 +3,20 @@
 import { assertProposalShape } from '@agoric/zoe/src/contractSupport/index.js';
 
 import '@agoric/zoe/exported.js';
+import { AmountMath } from '@agoric/ertp';
 
 /**
  * @param {ZCF} zcf
- * @param {(brandIn: Brand, brandOut: Brand) => VPoolWrapper<unknown>} provideVPool
+ * @param {(brandIn: Brand, brandOut: Brand) => VPoolWrapper<DoublePoolInternalFacet>} provideVPool
```

Comment:
provideVPool can provide either singlePool or doublePool, though no one should care. I'd make it `VPoolWrapper<unknown>`, which allows future extensions and declares that we don't care.